sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/FileBasedDataSourceTest.scala

```diff
@@ -52,7 +54,8 @@ private[sql] trait FileBasedDataSourceTest extends SQLTestUtils {
       f(spark.read.format(dataSourceName).load(path.toString))
     }
     if (testVectorized) {
-      withSQLConf(vectorizedReaderEnabledKey -> "true") {
+      withSQLConf(vectorizedReaderEnabledKey -> "true",
+          vectorizedReaderNestedEnabledKey -> "true") {
```

Comment:
Could you revise the test framework to have a complete test coverage for both `true` and `false`, @sunchao ?

Reply:
Sure, good idea.